Tools/CMake/AMReXFlagsTargets.cmake

```diff
@@ -99,12 +99,20 @@ target_compile_options( Flags_CXX_REQUIRED
 
 # Currently can't make this a generator expression as amrex_evaluate_genex fails
 # to parse it and propagate the flags to cuda
-if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC" AND CXX_COMPILER_VERSION VERSION_LESS 19.26)
-   target_compile_options( Flags_CXX_REQUIRED
-      INTERFACE
-      $<${_cxx_msvc}:/experimental:preprocessor /Zc:preprocessor>
-   )
+if (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+   if (CXX_COMPILER_VERSION VERSION_LESS 19.26)
+      target_compile_options( Flags_CXX_REQUIRED
+         INTERFACE
+         $<${_cxx_msvc}:/experimental:preprocessor>
```

Comment:
is the generator expression still needed when we have a `CMAKE_CXX_COMPILER_ID STREQUAL "MSVC"` guard?

Reply:
No it's not

Reply:
But don't we `if` the `CMAKE_CXX_COMPILER_ID` for MSVC as well (line 102)?

Reply:
The IF statement is a workaround for a deficiency in the current genex evaluation function. It will be removed when we fix that. The generator expression is what we want to keep so there is no point in modifying it now.